src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -708,8 +708,26 @@ protected void OnExited(object sender, EventArgs e)
                 //
                 stdInWriter.StopWriting();
             }
-            PSRemotingTransportException psrte = new PSRemotingTransportException(PSRemotingErrorId.IPCServerProcessExited,
-                                RemotingErrorIdStrings.IPCServerProcessExited);
+
+            int exitCode;
+            string exitMessage;
+            try
+            {
+                var jobProcess = (Process)sender;
+                exitCode = jobProcess.ExitCode;
+                exitMessage = exitCode == 0 ? jobProcess.StandardOutput.ReadToEnd() : jobProcess.StandardError.ReadToEnd();
+            }
+            catch (Exception exception)
+            {
+                exitCode = -1;
+                exitMessage = StringUtil.Format("<Unable to read streams from job process for reason '{0}'>", exception.Message);
```

Comment:
This error message string should also go into the resx file for eventual localization.